src/upnp.cpp

```diff
@@ -796,6 +799,8 @@ void upnp::next(rootdevice& d, port_mapping_t const i)
 void upnp::update_map(rootdevice& d, port_mapping_t const i)
 {
 	TORRENT_ASSERT(is_single_thread());
+	std::shared_ptr<upnp> me(self());
```

Comment:
as far as I can tell, all async operations in here correctly captures a refcounted shared_ptr to the upnp object. I don't see a way for the last reference to be released in here.